physlr/physlr.py

```diff
@@ -1800,10 +1994,15 @@ def index_minimizers_in_backbones(backbones, bxtomxs):
         for tid, path in enumerate(progress(backbones)):
             for pos, u in enumerate(path):
                 if Physlr.args.mx_type == "unsplit":
-                    if u not in bxtomxs:
+                    iter_c = 0
+                    while u not in bxtomxs:
                         u = u.rsplit("_", 1)[0]
+                        if iter_c > 20:
```

Comment:
What is this variable `iter_c` for?

Reply:
we keep removing `_`s from the name of barcode until we find the barcode in the dictionary. Normally, we definitely find it after a couple of iterations but I've put this to make sure if someone uses a wrong input file we do not trap in an infinite loop!

Reply:
Aren't we just changing the molecule IDs to "barcode_num"? I know we made that assumption in Johnathan's split-minimizers C++ code

Reply:
I double checked with Johnathan and we will postpone this PR to make some changes for the molecule id

Reply:
Sounds good -- yeah I think it'll make things easier to keep the molecule IDs in that form regardless of the number of rounds of molecule splitting.